src/os/windows/mod.rs

```diff
@@ -523,15 +485,26 @@ pub const LOAD_LIBRARY_SEARCH_USER_DIRS: LOAD_LIBRARY_FLAGS = consts::LOAD_LIBRA
 /// See [flag documentation on MSDN](https://docs.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-loadlibraryexw#parameters).
 ///
 /// [Remarks]: https://docs.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-loadlibraryexw#remarks
-pub const LOAD_WITH_ALTERED_SEARCH_PATH: LOAD_LIBRARY_FLAGS = consts::LOAD_WITH_ALTERED_SEARCH_PATH;
+pub const LOAD_WITH_ALTERED_SEARCH_PATH: LOAD_LIBRARY_FLAGS = 0x00000008;
 
 /// Specifies that the digital signature of the binary image must be checked at load time.
 ///
 /// See [flag documentation on MSDN](https://docs.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-loadlibraryexw#parameters).
-pub const LOAD_LIBRARY_REQUIRE_SIGNED_TARGET: LOAD_LIBRARY_FLAGS = consts::LOAD_LIBRARY_REQUIRE_SIGNED_TARGET;
+pub const LOAD_LIBRARY_REQUIRE_SIGNED_TARGET: LOAD_LIBRARY_FLAGS = 0x00000080;
 
 /// Allow loading a DLL for execution from the current directory only if it is under a directory in
 /// the Safe load list.
 ///
 /// See [flag documentation on MSDN](https://docs.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-loadlibraryexw#parameters).
-pub const LOAD_LIBRARY_SAFE_CURRENT_DIRS: LOAD_LIBRARY_FLAGS = consts::LOAD_LIBRARY_SAFE_CURRENT_DIRS;
+pub const LOAD_LIBRARY_SAFE_CURRENT_DIRS: LOAD_LIBRARY_FLAGS = 0x00002000;
+
+#[link(name="kernel32")]
```

Comment:
If you want to drop `windows-sys` I'd still recommend using `windows-targets`. See https://github.com/Amanieu/parking_lot/pull/378

Reply:
Thanks for bringing this up and taking time to review this PR!

I don’t entirely follow what problem this crate solves from reading the linked issue. Is it that the community doesn't always expect the linker to have access to ambient import libs, some form of which `windows-targets` (or rather – its dependencies) provides?

I guess that's reasonable enough, though it still leaves a bad taste in my mouth that using `windows-targets` is going to be slightly under a megabyte of dependencies which I’ll have to keep up with if I don't want this particular crate to eventually become the only crate that is responsible for pulling that megabyte in. Avoiding this ongoing maintenance effort was pretty much my rationale for migrating away from `windows-sys` in the first place.

---

So, I guess I have two overarching goals:

* I want to effectively never think about, or be pressured to, release new versions of this crate solely for dependency updates;
* I want this crate to be minimal in resources it requires to build (including network bandwidth.)

From what I can tell there are only a couple of plausible solutions compatible with these goals:

* `windows-targets` stops releasing semver incompatible changes as often (or claims that a bound similar to `>=0.48.0 && <1.0.0` is really likely to continue building and satisfying the above goals;)
* I somehow generate import libraries for all the relevant windows targets containing just the required APIs and ship them with libloading (I imagine those could be quite small.)

I wonder if this generating the import libraries thing could be a windows-rs tooling feature request (if it doesn’t yet exist?)

Reply:
Long term the recently stabilized `raw-dylib` should make this a thing of the past. Short term the `windows-targets` crate should very rarely release a semver-incompatible update. You can of course leave the code as is but that actually has a far bigger implicit dependency on the Windows SDK but may be "good enough" for most dependents. Here's an explainer on the `windows-targets` crate:

https://kennykerr.ca/rust-getting-started/understanding-windows-targets.html